### PR TITLE
#159054008 User Should Receive Descriptive Validation Errors.

### PR DESCRIPTION
**Pull request by dessHub**
**What does this PR do?**
. Implements descriptive validation error messages.

**How To Manually Test?**
1. Pull `ft-validators-159054008` branch.
  `git pull origin ft-validators-159054008`
2. Checkout to `ft-validators-159054008` branch
  `git checkout ft-validators-159054008`
3. Start the server
   `python manage.py runserver`
4. Test the endpoints on postman.

**Affected Endpoints:**
   POST \api\users
 
    {
      "user":{
        "username": "testuser",
        "email": "info",
        "password": "Pass123."
      }
    }

    RESPONSE
    {
        "errors": {
            "email": [
                "Enter a valid email address."
            ]
        }
    }

**What are the relevant pivotal tracker stories?**

[#159054008](https://www.pivotaltracker.com/story/show/159054008) Users should receive descriptive signup/Registration validation errors

### CODE REVIEW

File: authors/apps/authentication/tests/test_validators.py [hunk 1]
```diff
@@ -67,3 +67,44 @@ def test_username_cannot_be_register_twice(self):
         errors = json.loads(response.content).get("errors")
         self.assertEqual(errors['username'][0], "user with this username already exists.")
         self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+        
+    def test_password_cannot_be_lass_than_8_characters(self):
```
Comment (denniswanjiru) [inline authors/apps/authentication/tests/test_validators.py]: fix typo lass to less

File: authors/apps/authentication/tests/test_validators.py [hunk 2]
```diff
@@ -67,3 +67,44 @@ def test_username_cannot_be_register_twice(self):
...
+            reverse("authentication:registration"),
+            user,
+            format='json')
+
+        response.render()
+        errors = json.loads(response.content).get("errors")
+        self.assertEqual(errors['password'][0], "Password should be atleats 8 characters.")
+        self.assertEqual(response.status_code, status.HTTP_400_BAD_REQUEST)
+
+    def test_password_Not_being_alphanumeric(self):
```
Comment (denniswanjiru) [inline authors/apps/authentication/tests/test_validators.py]: I think you should not capitalize words in snake_case naming convention

**Comment by jacksono** [comment]
@dessHub are you still making changes on this PR?

**Comment by dessHub** [comment]
@jacksono Done updating the PR

**Comment by jacksono** [comment]
@dessHub from testing on my machine i have noticed that a user can register with a password is that not alphanumeric. In the example below, the password does not contain a number but the user gets registered
screen shot 2018-08-01 at 23 11 13
 Also I have merged in a new PR so you need to pull the latest changes from develop into your branch.

**Comment by dessHub** [comment]
@jacksono Will look into it right away.

**Comment by dessHub** [comment]
@jacksono I have fixed the bug.

**Comment by Duncanian** [comment]
@dessHub , I tested it with a wrong email and got the correct error message. It works as expected

**Review by jacksono — APPROVED** [review]

**Pull request merged**